main/2024/CG-06.md

```diff
@@ -93,10 +93,1859 @@ scheduled times if previous sessions end early.
+
+AK: This is a shared community. We thought strings might work well in the language, and people outside the web said, “we don’t want those string things in our nice pure wasm that’s not on the web.” Proposals like this are naturally going to reflect compromises between different parts. It didn’t sound like having a thread creation was a strict requirement. And we can keep the door open to adding a thread creation instruction in the future.
+
+BT: I don’t buy that argument. From the beginning, people have always been asking for a thread creation primitive. Why is there not a way to do that in the bytecode?
+
+CW: If you had a function to import, would people still be complaining about the lack of a bytecode?
+
+AK: Kind of the same reason we didn’t do it in the original threads/atomics proposal, it still isn’t necessary.
+
+TS: From an outside the Web perspective, I would have preferred for there to be a bytecode for creating threads, and I still hope we get there. I also do see how there is no consensus for how to do this both outside the Web and on the Web, so TL’s proposal is the best compromise we can get for now. We have a way in the CM to do this as one way to do it, and there are nonstandard ways. Extremely important to keep Web and CM aligned so that we are forward compatible with an eventual bytecode that works for both.
+
+MB: What do architectures support? No hardware architectures have a create-thread instruction. There’s a scheduler, an OS, lots of machinery. Everyone does it by calling into a host environment. It’s not a trivial hardware-like operation.
+
+TS: That also applies to Wasm GC, stack switching, and many other things.
+
+CW: RH, can you talk about thread local globals?
+
+RH: Have discussed that a lot in issues, and we don’t have a lot of time right now, so redirecting with a new question: What are your thoughts on how to proceed on the issue? 
+
+TL: Yes, we’ve had a lot of discussions and they’ve been productive, and they’ve stopped. So we are in a position to wait until we have a prototype and some performance numbers.
+
+RH: Are you interested in implementing a different approach so that you can compare them? Especially if the first implementation is obviously slow?
+
+TL: If the first implementation is fast, maybe that’s good enough?
+
+LW: numbers will help with “can tls be fast?” it won’t help with the shared-to-unshared edges discussion.
+
+TL, RH: Agreed.
+
+RH: It’s not just a big project in SpiderMonkey, but also in Gecko, to get right, and that’s concerning.
+
+TL: We need to wait for more implementor feedback. What do they need, what can they get away with? Evaluate the tradeoff in light of that data.
+
+RH: Toolchains are going to want the stronger semantics.
+
+TL: We can push back and ask for feedback;“what if you can’t have the stronger semantics?”
+#### Memory64 (Sam Clegg)
+
+TODO: link to slides slides 
```

Comment:
@sbc100 do you have a link to your slides?

Reply:
Friendly bump @sbc100

Reply:
Sorry, I'm was sure I posted this already: https://docs.google.com/presentation/d/1LRIIOUHZoTm127fnM-BuUDjVLOgUUzvqISJgiI_fWeY/edit?usp=sharing